Fix scrollbar not always being on top

TOULR opened the pull request:
Set the elevation of the scrollbar to 1 borderless, so that the blue outline is no longer above the scrollbar.

Closes #19875

Release Notes:

- N/A

TOULR committed:
Set the elevation of the scrollbar to 1 borderless, so that the blue outline is no longer above the scrollbar.
This should fix #19875

SomeoneToIgnore reviewed (dismissed):
Please post the screenshots of the state before and after.

TOULR commented on crates/project_panel/src/project_panel.rs:
```diff
@@ -2982,8 +2982,9 @@ impl ProjectPanel {
                 .right_1()
                 .top_1()
                 .bottom_1()
-                .w(px(12.))
```
I changed the width because otherwise there would be a gap between the right side of the scroll handle and where the blue border continues:
![grafik](https://github.com/user-attachments/assets/8119ddaf-c3db-466b-9809-a81f60c5dc23)
In my opinion it looks better like this:
![grafik](https://github.com/user-attachments/assets/46d48b02-8cb7-4a8a-94b4-17d31315ca42)
but i think it's a matter of taste

TOULR replied:
Maybe this could be solved in a different way, but i don't know how, this is my first contribution to this project

osiewicz commented:
It looks like this change causes the background of scrollbar show up unconditionally. We probably need render_vertical_scrollbar to return None if `Scrollbar::vertical` returns None as well.

TOULR committed:
Better solution to #19875
The blend function is used to blend the scrollbar color (scrollbar_thumb_background) on top of the background color (surface background). This way the color of the scrollbar stays the same, but it is no longer transparent, so that the blue border of selected files can't shine through and appear as if it was above the scrollbar.

TOULR commented:
I now found a possibly better solution to the problem with the blue border appearing on top of the scrollbar:
the blue border was appearing to be over the scrollbar because the color of the scrollbar has some transparency, but when i use the blend function to blend the scrollbar color on top of the background color, the color stays the same but the blue border of selected files no longer shines through
It now looks like this:
![grafik](https://github.com/user-attachments/assets/5a44550f-08a6-4af3-a89e-5cc1cf94b92d)

And now the background of the scrollbar doesn't appear when there is no scrollbar:
![grafik](https://github.com/user-attachments/assets/8e26bbfb-390e-4980-ad65-36df3faeba87)

osiewicz reviewed (approved):
Sweet, that makes sense! It seems like we're good to merge then, sans code formatting.

notpeter committed:
Fix formatting.
